src/commands/commerce/extension/register.ts

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+// What: Implemented a register command which registers an apex class via EPN to a web store
+
+// Why: Store admins will now be able to run through the full workflow of registering their Apex class from their command line
+// rather than via workbench or running custom apex code
+
+// How: command triggers query to retrieve apex class -> inserts fields into RegisteredExternalServiceTable -> returns a json response of current inserted record
+
+import { flags, SfdxCommand } from '@salesforce/command';
+import { Messages, Org, SfdxError } from '@salesforce/core';
+import { forceDataRecordCreate, forceDataSoql } from '../../../lib/utils/sfdx/forceDataSoql';
+import { StatusFileManager } from '../../../lib/utils/statusFileManager';
+
+Messages.importMessagesDirectory(__dirname);
+
+const TOPIC = 'extension';
+const CMD = `commerce:${TOPIC}:register`;
+const msgs = Messages.loadMessages('@salesforce/commerce', 'store');
+
+export class RegisterExtension extends SfdxCommand {
+    public static readonly requiresUsername = true;
+    // Setting specific flags
+    public static description = msgs.getMessage('extension.register.cmdDescription');
+    public static example = [`sfdx ${CMD} --registered-extensionname --extensionpointname --apex-classname`];
+
+    public static flagsConfig = {
+        'registered-extensionname': flags.string({
```

Comment:
nit: would we consider combining the words (no kebab case) for the flags to keep it consistent with the other flags in the sfdx cli command?

Reply:
my team and I took some time to review and gave feedback that kebab case should be used for all the fields to be consistent

`-extensionpointname` ->  `--extension-point-name`
`--registered-extensionname` -> `--registered-extension-name`
`--apex-classname` -> `--apex-class-name`

let me know if that's okay!